OpenSim/Sandbox/pseudoKinematicsReporting.cpp

```diff
@@ -0,0 +1,51 @@
+
+#include <OpenSim/OpenSim.h>
+
+main() {
+    // A study is the top level component that contains the model and other 
+    // computational components. 
+    Study kinematicsStudy;
+
+    Model model("subject_01.osim");
+    // previous simulation result with the same model
+    StatesTrajectory states("states.ost");
+
+    kinematicsStudy.addComponent(model);
+
+    OutputReporter bodyKinematics;
+    // Replace exactly what body kinematics analysis does
+    for (const auto& body : model.getComponents<Body>()) {
+        bodyKinematics.addOutput(body.getOutput("transfom"));
+    }
+    kinematicsStudy.addReporter(bodyKinematics);
+
+
+    // Print out a list of all the available outputs from Frames in the model
+    for (const auto& frame : model.getComponents<Frame>()) {
+        for (const auto& output : frame.getOutputs()) {
+            cout << "Frame " frame.getName();
+            cout << " output: " << output.getName() << endl;
```

Comment:
Perhaps all Components can have a method like `getPath()` that would concatenate the frame name and output name as a convenience. To disambiguate with filesystem path, the method could also be called `getComponentPath()`.